lib/matplotlib/__init__.py

```diff
@@ -830,12 +830,17 @@ def rc_params_from_file(fname, fail_on_error=False, use_default_template=True):
     rcParamsOrig = RcParams(rcParams.copy())
     # This also checks that all rcParams are indeed listed in the template.
     # Assigning to rcsetup.defaultParams is left only for backcompat.
-    defaultParams = rcsetup.defaultParams = {
-        # We want to resolve deprecated rcParams, but not backend...
-        key: [(rcsetup._auto_backend_sentinel if key == "backend" else
-               rcParamsDefault[key]),
-              validator]
-        for key, validator in rcsetup._validators.items()}
+    defaultParams = {}
+    for key, validator in rcsetup._validators.items():
+        if key == 'backend':
+            defaultParams['backend'] = rcsetup._auto_backend_sentinel
+        elif key in rcParamsDefault.keys():
+            defaultParams[key] = validator
```

Comment:
I'm a bit confused that before `defaultParams` was a mapping `key: [default_value, validator]` and now it's `key: validator`.
Both don't make sense to me. I'd expect `key: default_value`. But maybe that's just my before-first-coffee brain.

Reply:
Ooops, well, it is certainly a mystery to me what this is even supposed to do if this big error didn't kill the test suite.   A search in the repo turns up `defaultParams` only gets used here, and nowhere else.  Removing this whole block of code leads to no changes so far as I can tell.  So I am completely mystified why this code even exists.

Reply:
Just pushed a new commit that cuts it all.  So far tests are fine.  So again, what is this supposed to do (he sheepishly asks of a PR change he approved #15029)